### PR TITLE
[Collections] SQLite optimizations

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
Add/delete/refresh package collections lead to frequent deletes in SQLite. This causes the database file size to grow very quickly when we have a huge collection.

Modification:
- Run `VACUUM` after SQLite delete. This repacks the database file and helps reduce its size.
- Don't update FTS tables unless the collection's packages have changed. This helps reduce the number of deletes.

Related to rdar://77077510

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]

File: Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift [hunk 1]
```diff
@@ -126,28 +126,38 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
...
-                    let data = try self.encoder.encode(collection)
-
-                    let bindings: [SQLite.SQLiteValue] = [
-                        .string(collection.identifier.databaseKey()),
-                        .blob(data),
-                    ]
-                    try statement.bind(bindings)
-                    try statement.step()
-                }
+            self.get(identifier: collection.identifier) { getResult in
```
Comment (tomerd) [inline Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:129]: nit and not sure if its worth it, but maybe an "exists" API can be faster than a full get

Comment (yim-lee) [reply Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:129]: We do need the `Collection` to compare packages.

Comment (tomerd) [reply Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:129]: missed that... 👍

**Review by neonichu — APPROVED** [review]

**Pull request merged**